language/sygus.tex

```diff
@@ -1290,116 +1284,95 @@ \subsection{Declaring Oracle Functional Symbols}
 
 \begin{itemize}
 \item $\paren{\iooracledeckwd\mbox{ }F\mbox{ } N}$
 
-This declares an input-output oracle for function $F$. 
+This command declares an input-output oracle for function $F$. 
 It is syntactic sugar for:
 
-
-$\paren{\constraintoraclekwd\mbox{ } \paren{\paren{x_1\mbox{ }\sigma_1} \ldots \paren{x_n\mbox{ }\sigma_n}}\mbox{ }
-\paren{\paren{x \mbox{ }\sigma}}
-\paren{{\tt=}\mbox{ } \paren{F \mbox{ } x_1 \ldots x_n}\mbox{ }  x}\mbox{ }N
-}
-$
-
+\begin{lstlisting}[language=SyGuS-Desugar]
+($\constraintoraclekwd$ (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)) (($x$ $\sigma$)) (= ($F$ $x_1$ $\ldots$ $x_n$) $x$) $N$)
+\end{lstlisting}
 
 \item$\paren{\poswitnessoracledeckwd\mbox{ }F\mbox{ } N} $
 
-This declares a positive witness oracle for synthesis function $F$ and is syntactic sugar for:
+This command declares a positive witness oracle for synthesis function $F$.
+It is syntactic sugar for:
 
-$\paren{\constraintoraclekwd\mbox{ } 
-\mbox{ }
-\paren{\paren{x_1\mbox{ }\sigma_1} \ldots \paren{x_n\mbox{ }\sigma_n} \paren{x \mbox{ }\sigma}}\mbox{ }
-\paren{{\tt=}\mbox{ } \paren{F \mbox{ } x_1 \ldots x_n}\mbox{ }  x}\mbox{ }N
-}
-$
+\begin{lstlisting}[language=SyGuS-Desugar]
+($\constraintoraclekwd$ (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)) (($x$ $\sigma$)) (= ($F$ $x_1$ $\ldots$ $x_n$) $x$) $N$)
+\end{lstlisting}
 
 \item$\paren{\negwitnessoracledeckwd\mbox{ }F\mbox{ } N}$
 
-This declares a negative witness oracle for synthesis function $F$ and  is syntactic sugar for:
-
-% $\paren{\oracledeckwd\mbox{ }S\mbox{ }
-% \paren{}
-% \paren{\paren{x_1\mbox{ }\sigma_1} \ldots \paren{x_n\mbox{ }\sigma_n} \paren{x \mbox{ }\sigma}}\mbox{ }
-% }$
-
-$\paren{\constraintoraclekwd\mbox{ } 
-\mbox{ }
-\paren{\paren{x_1\mbox{ }\sigma_1} \ldots \paren{x_n\mbox{ }\sigma_n} \paren{x \mbox{ }\sigma}}\mbox{ }
-\paren{{\tt not} \mbox{ } \paren{{\tt=} \mbox{ } \paren{F \mbox{ } x_1 \ldots x_n}\mbox{ }  x}} \mbox{ } N}
-$
+This command declares a negative witness oracle for synthesis function $F$.
+It is syntactic sugar for:
 
+\begin{lstlisting}[language=SyGuS-Desugar]
+($\constraintoraclekwd$ (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)) (($x$ $\sigma$)) (not (= ($F$ $x_1$ $\ldots$ $x_n$) $x$)) $N$)
+\end{lstlisting}
 
 \item$\paren{\memoracledeckwd\mbox{ }F\mbox{ } N} $
 
-This declares a membership-query oracle for synthesis function $F$ and is syntactic sugar for:
-
-
-$\paren{\constraintoraclekwd \mbox{ }
-\paren{\paren{x_1\mbox{ }\sigma_1} \ldots \paren{x_n\mbox{ }\sigma_n} \paren{x \mbox{ }\sigma}}\mbox{ }
-\paren{\paren{R \mbox{ } \sbool}}\mbox{ } 
-%
-\paren{{\tt=} \mbox{ } \paren{{\tt=}\mbox{ } \paren{F \mbox{ } x_1 \ldots x_n } \mbox{ }  x} \mbox{ } R}  \mbox{ }N}
-$
+This command declares a membership-query oracle for synthesis function $F$.
+It is syntactic sugar for:
 
-% 
+\begin{lstlisting}[language=SyGuS-Desugar]
+($\constraintoraclekwd$
+  (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$))
+  (($x$ $\sigma$) ($R$ Bool))
```

Comment:
The parenthesis have changed here, should have
input: (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)($x$ $\sigma$))
output: (($R$ Bool))

Reply:
```suggestion
  (($x_1$ $\sigma_1$)$\ldots$($x_n$ $\sigma_n$)($x$ $\sigma$))
   (($R$ Bool))
```

Reply:
Thanks, fixed as suggested.